src/Pages/MainPage/Components/TaskList.tsx

```diff
@@ -36,6 +70,25 @@ const TaskList: React.FC<TaskListProps> = ({
     toggleCompletedTask(id);
   };
 
+  const handleEditChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    const { name, value } = e.target;
+    setEditFormState((prev) => ({ ...prev, [name]: value }));
+  };
```

Comment:
You can look into usage of form and validation libraries such as react hook form and zod.

Can create a ticket to hook that up later